In [15]:
import pandas as pd
import os
import requests
from datetime import datetime, timedelta
import requests
from bs4 import BeautifulSoup as bs
import data.publisher as pb
import data.temp as temp
import json
import data.tools as tools

In [18]:
test=tools.get_title_contents(input())

In [41]:
class_name=pb.news[0]

In [40]:
soup.select_one(pb.news[0]["title"]).text


'악수하는 이정미 대표와 정진석 비상대책위원장'

In [42]:
soup.select_one(class_name["title"]).text

'악수하는 이정미 대표와 정진석 비상대책위원장'

In [45]:
contents_text = str(soup.select_one(class_name["contents"]))
contents_text=[i for i in contents_text.split("<br/>") if len(i) >1 and "class=" not in i and "id="not in i and "span" not in i and "strong" not in i]
new_news=[]
remover=["\n","\t","</div>","\'"]
for i in range(len(contents_text)):
    line=contents_text[i].strip()
    for j in remover:
        line=line.replace(j,"")
    new_news.append(line)
seperator=","
new_news=seperator.join(new_news)

In [46]:
new_news

'(서울=연합뉴스) 하사헌 기자 = 정의당 이정미 대표가 10일 오전 서울 여의도 국회에서 국민의힘 정진석 비상대책위원장을 예방해 악수하고 있다. 2022.11.10,toadboy@yna.co.kr,'

In [47]:
soup.select_one(class_name["press"][0]).get(class_name["press"][1])

'연합뉴스'

In [48]:
soup.select_one(class_name["img"][0]).get(class_name["img"][1])

'https://imgnews.pstatic.net/image/001/2022/11/10/PYH2022111005620001301_P4_20221110103811966.jpg?type=w647'

In [51]:
soup.select_one(class_name["reporter"]).text

'하사헌 기자'

In [ ]:
pb.news[1]["contents"]

In [61]:
soup.select_one(pb.news[1]["press"][0]).text

'머니투데이'

In [63]:
contents_text

[]

In [132]:
time=soup.select_one(pb.news[1]["time"])


In [133]:
time

<span class="txt_info">입력 <span class="num_date">2022. 11. 9. 15:15</span></span>

In [123]:
contents_text= [i.text.strip() for i in contents_text]

In [124]:
contents_text

['2023학년도 대학수학능력시험이 오는 17일 치러지는 가운데, 코로나19에 확진된 수험생은 병원이 아닌 별도 시험장에서 시험을 볼 수 있습니다.',
 '확진자에 대한 7일의 격리 의무 기간을 고려해 오는 11일 이후 코로나19 확진 판정을 받은 수험생은 지체 없이 관할 교육청에 알려야 수능 응시에 필요한 지원을 받을 수 있습니다.',
 '당국은 일반 수험생과 분리된 별도 고사장을 마련해 따로 시험을 볼 수 있도록 했습니다.',
 '임숙영 방역대책본부 상황총괄단장은 오늘(9일) 코로나19 중앙재난안전대책본부 정례 브리핑에서 "수능에 응시한 수험생은 확진을 받았다고 하더라도 시험을 볼 수 있다"며 "확진 수험생의 경우 별도로 마련된 고사장에서 시험을 본다"고 밝혔습니다.',
 '수험생과 수험생 가족은 코로나19 의심 증상이 나타나면 즉시 인근 호흡기환자진료센터에 방문해 신속항원검사를 받아야 합니다.',
 '검사를 받을 때는 검사기관에 본인이 수능 응시생이라는 점을 알려야 합니다.',
 '만약 확진자 중 입원 치료가 필요한 수험생은 별도의 병원 시험장에서 시험을 볼 수 있습니다.',
 "또 격리 대상은 아니지만 수능 당일 2~3차례 체온 검사에서 37.5도 이상 열이 나는 수험생의 경우엔 '유증상 수험생'으로 분류돼, 일반 시험장에 따로 마련된 분리 시험실에서 시험을 봐야 합니다.",
 '모든 수험생은 마스크를 써야 하고 휴대전화나 블루투스 이어폰 등 전자기기 반입은 금지되는데, 반입 금지 물품을 가지고 들어간 경우 1교시 전 감독관에게 이를 제출해야 합니다.',
 '시계는 전자식 화면 표시나 통신 기능이 있는 것은 모두 사용할 수 없고, 아날로그 시계만 허용됩니다.',
 '점심시간에는 종이 칸막이를 직접 설치한 뒤 식사하면 됩니다.',
 '특히 4교시 탐구 영역 때는 본인이 선택한 과목을 순서대로 응시하면 되는데, 동시에 2과목 문제지를 올리거나 풀면 부정행위로 처리돼 주의해야 합니다.']

In [125]:
new_news=[]
remover=["\n","\t","</div>","\'"]
for i in range(len(contents_text)):
    line=contents_text[i].strip()
    for j in remover:
        line=line.replace(j,"")
    new_news.append(line)
seperator=","
seperator.join(new_news)

'2023학년도 대학수학능력시험이 오는 17일 치러지는 가운데, 코로나19에 확진된 수험생은 병원이 아닌 별도 시험장에서 시험을 볼 수 있습니다.,확진자에 대한 7일의 격리 의무 기간을 고려해 오는 11일 이후 코로나19 확진 판정을 받은 수험생은 지체 없이 관할 교육청에 알려야 수능 응시에 필요한 지원을 받을 수 있습니다.,당국은 일반 수험생과 분리된 별도 고사장을 마련해 따로 시험을 볼 수 있도록 했습니다.,임숙영 방역대책본부 상황총괄단장은 오늘(9일) 코로나19 중앙재난안전대책본부 정례 브리핑에서 "수능에 응시한 수험생은 확진을 받았다고 하더라도 시험을 볼 수 있다"며 "확진 수험생의 경우 별도로 마련된 고사장에서 시험을 본다"고 밝혔습니다.,수험생과 수험생 가족은 코로나19 의심 증상이 나타나면 즉시 인근 호흡기환자진료센터에 방문해 신속항원검사를 받아야 합니다.,검사를 받을 때는 검사기관에 본인이 수능 응시생이라는 점을 알려야 합니다.,만약 확진자 중 입원 치료가 필요한 수험생은 별도의 병원 시험장에서 시험을 볼 수 있습니다.,또 격리 대상은 아니지만 수능 당일 2~3차례 체온 검사에서 37.5도 이상 열이 나는 수험생의 경우엔 유증상 수험생으로 분류돼, 일반 시험장에 따로 마련된 분리 시험실에서 시험을 봐야 합니다.,모든 수험생은 마스크를 써야 하고 휴대전화나 블루투스 이어폰 등 전자기기 반입은 금지되는데, 반입 금지 물품을 가지고 들어간 경우 1교시 전 감독관에게 이를 제출해야 합니다.,시계는 전자식 화면 표시나 통신 기능이 있는 것은 모두 사용할 수 없고, 아날로그 시계만 허용됩니다.,점심시간에는 종이 칸막이를 직접 설치한 뒤 식사하면 됩니다.,특히 4교시 탐구 영역 때는 본인이 선택한 과목을 순서대로 응시하면 되는데, 동시에 2과목 문제지를 올리거나 풀면 부정행위로 처리돼 주의해야 합니다.'

In [117]:

contents_text= [i.text.strip() for i in contents_text]


In [111]:
contents_text


['2023학년도 대학수학능력시험이 오는 17일 치러지는 가운데, 코로나19에 확진된 수험생은 병원이 아닌 별도 시험장에서 시험을 볼 수 있습니다.',
 '확진자에 대한 7일의 격리 의무 기간을 고려해 오는 11일 이후 코로나19 확진 판정을 받은 수험생은 지체 없이 관할 교육청에 알려야 수능 응시에 필요한 지원을 받을 수 있습니다.',
 '당국은 일반 수험생과 분리된 별도 고사장을 마련해 따로 시험을 볼 수 있도록 했습니다.',
 '임숙영 방역대책본부 상황총괄단장은 오늘(9일) 코로나19 중앙재난안전대책본부 정례 브리핑에서 "수능에 응시한 수험생은 확진을 받았다고 하더라도 시험을 볼 수 있다"며 "확진 수험생의 경우 별도로 마련된 고사장에서 시험을 본다"고 밝혔습니다.',
 '수험생과 수험생 가족은 코로나19 의심 증상이 나타나면 즉시 인근 호흡기환자진료센터에 방문해 신속항원검사를 받아야 합니다.',
 '검사를 받을 때는 검사기관에 본인이 수능 응시생이라는 점을 알려야 합니다.',
 '만약 확진자 중 입원 치료가 필요한 수험생은 별도의 병원 시험장에서 시험을 볼 수 있습니다.',
 "또 격리 대상은 아니지만 수능 당일 2~3차례 체온 검사에서 37.5도 이상 열이 나는 수험생의 경우엔 '유증상 수험생'으로 분류돼, 일반 시험장에 따로 마련된 분리 시험실에서 시험을 봐야 합니다.",
 '모든 수험생은 마스크를 써야 하고 휴대전화나 블루투스 이어폰 등 전자기기 반입은 금지되는데, 반입 금지 물품을 가지고 들어간 경우 1교시 전 감독관에게 이를 제출해야 합니다.',
 '시계는 전자식 화면 표시나 통신 기능이 있는 것은 모두 사용할 수 없고, 아날로그 시계만 허용됩니다.',
 '점심시간에는 종이 칸막이를 직접 설치한 뒤 식사하면 됩니다.',
 '특히 4교시 탐구 영역 때는 본인이 선택한 과목을 순서대로 응시하면 되는데, 동시에 2과목 문제지를 올리거나 풀면 부정행위로 처리돼 주의해야 합니다.']

In [102]:
daum_text

['2023학년도 대학수학능력시험이 오는 17일 치러지는 가운데, 코로나19에 확진된 수험생은 병원이 아닌 별도 시험장에서 시험을 볼 수 있습니다.',
 '확진자에 대한 7일의 격리 의무 기간을 고려해 오는 11일 이후 코로나19 확진 판정을 받은 수험생은 지체 없이 관할 교육청에 알려야 수능 응시에 필요한 지원을 받을 수 있습니다.',
 '당국은 일반 수험생과 분리된 별도 고사장을 마련해 따로 시험을 볼 수 있도록 했습니다.',
 '임숙영 방역대책본부 상황총괄단장은 오늘(9일) 코로나19 중앙재난안전대책본부 정례 브리핑에서 "수능에 응시한 수험생은 확진을 받았다고 하더라도 시험을 볼 수 있다"며 "확진 수험생의 경우 별도로 마련된 고사장에서 시험을 본다"고 밝혔습니다.',
 '수험생과 수험생 가족은 코로나19 의심 증상이 나타나면 즉시 인근 호흡기환자진료센터에 방문해 신속항원검사를 받아야 합니다.',
 '검사를 받을 때는 검사기관에 본인이 수능 응시생이라는 점을 알려야 합니다.',
 '만약 확진자 중 입원 치료가 필요한 수험생은 별도의 병원 시험장에서 시험을 볼 수 있습니다.',
 "또 격리 대상은 아니지만 수능 당일 2~3차례 체온 검사에서 37.5도 이상 열이 나는 수험생의 경우엔 '유증상 수험생'으로 분류돼, 일반 시험장에 따로 마련된 분리 시험실에서 시험을 봐야 합니다.",
 '모든 수험생은 마스크를 써야 하고 휴대전화나 블루투스 이어폰 등 전자기기 반입은 금지되는데, 반입 금지 물품을 가지고 들어간 경우 1교시 전 감독관에게 이를 제출해야 합니다.',
 '시계는 전자식 화면 표시나 통신 기능이 있는 것은 모두 사용할 수 없고, 아날로그 시계만 허용됩니다.',
 '점심시간에는 종이 칸막이를 직접 설치한 뒤 식사하면 됩니다.',
 '특히 4교시 탐구 영역 때는 본인이 선택한 과목을 순서대로 응시하면 되는데, 동시에 2과목 문제지를 올리거나 풀면 부정행위로 처리돼 주의해야 합니다.']

In [113]:
seperator=","
seperator.join(contents_text)

'2023학년도 대학수학능력시험이 오는 17일 치러지는 가운데, 코로나19에 확진된 수험생은 병원이 아닌 별도 시험장에서 시험을 볼 수 있습니다.,확진자에 대한 7일의 격리 의무 기간을 고려해 오는 11일 이후 코로나19 확진 판정을 받은 수험생은 지체 없이 관할 교육청에 알려야 수능 응시에 필요한 지원을 받을 수 있습니다.,당국은 일반 수험생과 분리된 별도 고사장을 마련해 따로 시험을 볼 수 있도록 했습니다.,임숙영 방역대책본부 상황총괄단장은 오늘(9일) 코로나19 중앙재난안전대책본부 정례 브리핑에서 "수능에 응시한 수험생은 확진을 받았다고 하더라도 시험을 볼 수 있다"며 "확진 수험생의 경우 별도로 마련된 고사장에서 시험을 본다"고 밝혔습니다.,수험생과 수험생 가족은 코로나19 의심 증상이 나타나면 즉시 인근 호흡기환자진료센터에 방문해 신속항원검사를 받아야 합니다.,검사를 받을 때는 검사기관에 본인이 수능 응시생이라는 점을 알려야 합니다.,만약 확진자 중 입원 치료가 필요한 수험생은 별도의 병원 시험장에서 시험을 볼 수 있습니다.,또 격리 대상은 아니지만 수능 당일 2~3차례 체온 검사에서 37.5도 이상 열이 나는 수험생의 경우엔 \'유증상 수험생\'으로 분류돼, 일반 시험장에 따로 마련된 분리 시험실에서 시험을 봐야 합니다.,모든 수험생은 마스크를 써야 하고 휴대전화나 블루투스 이어폰 등 전자기기 반입은 금지되는데, 반입 금지 물품을 가지고 들어간 경우 1교시 전 감독관에게 이를 제출해야 합니다.,시계는 전자식 화면 표시나 통신 기능이 있는 것은 모두 사용할 수 없고, 아날로그 시계만 허용됩니다.,점심시간에는 종이 칸막이를 직접 설치한 뒤 식사하면 됩니다.,특히 4교시 탐구 영역 때는 본인이 선택한 과목을 순서대로 응시하면 되는데, 동시에 2과목 문제지를 올리거나 풀면 부정행위로 처리돼 주의해야 합니다.'

2023학년도 대학수학능력시험이 오는 17일 치러지는 가운데, 코로나19에 확진된 수험생은 병원이 아닌 별도 시험장에서 시험을 볼 수 있습니다.,확진자에 대한 7일의 격리 의무 기간을 고려해 오는 11일 이후 코로나19 확진 판정을 받은 수험생은 지체 없이 관할 교육청에 알려야 수능 응시에 필요한 지원을 받을 수 있습니다.,당국은 일반 수험생과 분리된 별도 고사장을 마련해 따로 시험을 볼 수 있도록 했습니다.,임숙영 방역대책본부 상황총괄단장은 오늘(9일) 코로나19 중앙재난안전대책본부 정례 브리핑에서 "수능에 응시한 수험생은 확진을 받았다고 하더라도 시험을 볼 수 있다"며 "확진 수험생의 경우 별도로 마련된 고사장에서 시험을 본다"고 밝혔습니다.,수험생과 수험생 가족은 코로나19 의심 증상이 나타나면 즉시 인근 호흡기환자진료센터에 방문해 신속항원검사를 받아야 합니다.,검사를 받을 때는 검사기관에 본인이 수능 응시생이라는 점을 알려야 합니다.,만약 확진자 중 입원 치료가 필요한 수험생은 별도의 병원 시험장에서 시험을 볼 수 있습니다.,또 격리 대상은 아니지만 수능 당일 2~3차례 체온 검사에서 37.5도 이상 열이 나는 수험생의 경우엔 유증상 수험생으로 분류돼, 일반 시험장에 따로 마련된 분리 시험실에서 시험을 봐야 합니다.,모든 수험생은 마스크를 써야 하고 휴대전화나 블루투스 이어폰 등 전자기기 반입은 금지되는데, 반입 금지 물품을 가지고 들어간 경우 1교시 전 감독관에게 이를 제출해야 합니다.,시계는 전자식 화면 표시나 통신 기능이 있는 것은 모두 사용할 수 없고, 아날로그 시계만 허용됩니다.,점심시간에는 종이 칸막이를 직접 설치한 뒤 식사하면 됩니다.,특히 4교시 탐구 영역 때는 본인이 선택한 과목을 순서대로 응시하면 되는데, 동시에 2과목 문제지를 올리거나 풀면 부정행위로 처리돼 주의해야 합니다.'

In [66]:
contents_text = str(soup.select_one(pb.news[1]["contents"]))
contents_text=[i for i in contents_text.split("<br/>") if len(i) >1 and "class=" not in i and "id="not in i and "span" not in i and "strong" not in i]
new_news=[]
remover=["\n","\t","</div>","\'"]
for i in range(len(contents_text)):
    line=contents_text[i].strip()
for j in remover:
    line=line.replace(j,"")
new_news.append(line)
seperator=","
seperator.join(new_news)

''

In [2]:
pd.read_csv("./save/links.csv")

link
0      https://n.news.naver.com/mnews/article/047/000...
1      https://n.news.naver.com/mnews/article/421/000...
2      https://n.news.naver.com/mnews/article/277/000...
3      https://n.news.naver.com/mnews/article/016/000...
4      https://n.news.naver.com/mnews/article/001/001...
...                                                  ...
11075  https://n.news.naver.com/mnews/article/001/001...
11076  https://n.news.naver.com/mnews/article/119/000...
11077  https://n.news.naver.com/mnews/article/468/000...
11078  https://n.news.naver.com/mnews/article/018/000...
11079  https://n.news.naver.com/mnews/article/119/000...

[11080 rows x 1 columns]

In [186]:
sample=pd.read_csv("./save/sample.csv")

In [187]:
sample

title  \
0                   "이태원 참사... 지자체 행사도 이렇게는 안 할 거예요"   
1                      한·뉴질랜드 태평양 대화 "기후변화 등 협력 파트너"   
2                [속보]대통령실 "종부세 부담 줄이는 다양한 방안 논의, 실천"   
3        [단독]“이태원 대응 복지부 직원 동원 여론 심상찮아”…조규홍 “가시적 조치”   
4                      대법 "공수처의 김웅 의원실 압수수색은 위법"(종합)   
5                       한덕수·이상민·윤희근 "책임 회피할 생각 없지만…"   
6  [10.29 참사] 윤건영 "왜 4시간 동안 쳐다만 보고 있었냐?"…"대통령이 경찰...   
7               [영상] 러, 최신 핵잠수함 성능 시험 완료…태평양함대 배치 예정   
8                  이재명, 오열하는 이태원 유족 손잡고 "미안하다, 죄송하다"   
9            [이태원 참사] 野 "한덕수·이상민 책임" vs 與 "경찰, 사상누각"   

                                             content  
0  ['“농민수당을 만든 사람”, “우리를 위해 밥도 굶어준 사람” “배지 없이도 구의...  
1  ["(서울=뉴스1) 이창규 기자 = 우리나라와 뉴질랜드 외교당국이 8일 '한·뉴질랜...  
2                                       ['\n</div>']  
3  ['[헤럴드경제=이상섭 기자] 조규홍 보건복지부 장관이 복지부 직원으로 추정되는 인...  
4  ['공수처 "참여권 범위 넓힌 선례로 이해…재판에는 영향 없어"</strong>',...  
5  ["한덕수 국무총리와 이상민 행정안전부 장관이 이태원 참사와 관련해 야당의 사퇴 압...  
6  ['박지원 "윤 대통령 4시간 동안 뭐하셨나?…국가위기관리센터 잘 작동했나?"', ...  
7  ['    보도에 따르면 세브마슈 조선소 미하일 부드니첸코 최고경영자(CEO)는 이...  
8  ['이재명 더불어민주당 대표는 8일 서울 용산구 이태원 파출소를 찾아 사고 당시 현...  
9  ["與, '정부 책임론' 선 긋기…제도 미비 지적</strong>", '(서울=연합...

In [183]:
url="https://v.daum.net/v/20221109151504839"

In [182]:

headers = {"user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.89 Safari/537.36"}
req = requests.get(url,headers=headers)
html=req.text
soup = bs(html, 'html.parser')


In [184]:
temp_time=soup.select_one(".num_date").text+":00"
date_time_str = temp_time
date_time_obj = datetime.datetime.strptime(date_time_str, '%Y. %m. %d. %H:%M:%S')
date_time_obj.strftime("%Y-%m-%d %H:%M:%S")



'2022-11-09 15:15:00'

'2022. 11. 9. 15:15:00'

'2022. 11. 9. 15:15:00'

In [176]:
import datetime

date_time_str = temp_time
date_time_obj = datetime.datetime.strptime(date_time_str, '%Y. %m. %d. %H:%M:%S')


In [181]:
date_time_obj.strftime("%Y-%m-%d %H:%M:%S")

'2022-11-09 15:15:00'

In [56]:
naver_press=soup.select_one(".media_end_head_top_logo_img").get("title")
naver_thumbnail=soup.select_one("#img1").get("data-src")
naver_time = soup.select_one(".media_end_head_info_datestamp_time").text
naver_reporter = soup.select_one(".media_end_head_journalist_name").text

AttributeError: 'NoneType' object has no attribute 'get'

In [57]:
daum_press=soup.select_one("#kakaoServiceLogo").get("data-tiara")
daum_thumbnail=soup.select_one(".thumb_g_article").get("src")
daum_time=soup.select_one(".info_view > span:nth-child(2)").text
daum_reporter=soup.select_one(".info_view > span:first-child").text

In [59]:
daum_thumbnail

'https://img3.daumcdn.net/thumb/R658x0.q70/?fname=https://t1.daumcdn.net/news/202211/10/moneytoday/20221110094129395caim.jpg'

In [58]:
daum_press

'언론사명'

In [52]:
soup.select_one(".info_view > span:nth-child(2)").text

'입력 2022. 11. 10. 09:38'

SyntaxError: invalid syntax. Perhaps you forgot a comma? (3426379995.py, line 3)

In [51]:
daum_reporter

'김도윤 기자'

In [15]:
body > div.container-doc > main > section > div > div.content-article > div.box_g.box_news_issue > ul > li:nth-child(1) > div > a

AttributeError: ResultSet object has no attribute 'select_one'. You're probably treating a list of elements like a single element. Did you call find_all() when you meant to call find()?

In [83]:
with open("./save/dailyCrawling.json", "r") as json_file:
    json_data = json.load(json_file)

'BOE 수석 이코노미스트 "펜데믹 양적완화는 실수였다"···공급확대 정책 실수 인정'

In [105]:
import urllib.request
from bs4 import BeautifulSoup
import os
from importlib.resources import read_text
import requests
import data.tools as tools
#네이버 기사 혹은 다음 기사 링크 인풋
news=tools.get_title_contents(input())

In [148]:
new_news=[]
remover=["\n","\t","</div>","\'"]
for i in range(len(news[1])):
    line=news[1][i].strip()
    for j in remover:
        line=line.replace(j,"")
    new_news.append(line)

seperator=","
seperator.join(new_news)


In [150]:
seperator=","
seperator.join(new_news)

'(서울=뉴스1) 이창규 기자 = 우리나라와 뉴질랜드 외교당국이 8일 한·뉴질랜드 태평양 대화를 열어 태평양 정세와 대(對)태평양 도서국 정책 및 개발협력 등에 대해 논의했다.,외교부에 따르면 서민정 아시아·태평양국 심의관과 조너선 오스틴 뉴질랜드 외교부 태평양국장은 이날 대화에서 "한국과 뉴질랜드는 인도·태평양 지역의 자유·평화·번영을 위해 함께 기여할 수 있는 파트너"라며 태평양 지역에서 양국이 협력할 수 있는 분야를 모색해가가기로 했다.,서 심의관은 특히 "한국 정부는 글로벌 중추 국가 비전에 따라 태평양 도서국과의 관계 강화를 중시하고 있다"며 "2050 푸른태평양대륙 전략 실현에도 기여할 수 있도록 태평양 도서국과의 협력을 확대해 나갈 예정"이라고 소개했다.,2050 푸른태평양대륙 전략은 지난 7월 태평양도서국포럼(PIF) 정상회의에서 공식 채택된 것으로 태평양 지역의 장기 개발 비전과 중점 분야 등을 담고 있다. 여기엔 △정치적 리더십과 지역주의 △사람 중심 개발 △평화·안보 △기후변화·재난 △해양·환경 △자원·경제개발 △기술·연결성 등이 포함된다.,오스틴 국장도 보건 시스템 강화와 기후변화 대응 등에서 우리 정부와 협력하고자 한다고 밝혔다.'

NameError: name 'old_string' is not defined